test/dlc_tests/sanity/test_ecr_scan.py

```diff
@@ -84,39 +84,55 @@ def get_minimum_sev_threshold_level(image):
     return "HIGH"
 
 
-def upload_allowlist_to_image_data_storage_bucket(
-    ecr_client_for_enhanced_scanning_repo, ecr_enhanced_repo_uri, allowlist_for_daily_scans
-):
+def upload_json_to_image_data_storage_s3_bucket(image_sha: str, upload_data: str, s3_filename: str):
     """
-    This method gets the unique identifier of the image from ecr and uses it as the path to
-    upload image's allowlisted vulnerabilities to the image-data-storage s3 bucket. This
-    information is used for the scanning dashboard to identify allowlisted vulnerabilities.
+    This method uses the unique identifier of the image as the path to upload the given image's `upload_data`
+    to the image-data-storage s3 bucket.
+    This information is used for the scanning dashboard to identify image allowlist information.
 
-    :param ecr_client_for_enhanced_scanning_repo: ecr boto3 client for image
-    :param ecr_enhanced_repo_uri: str Image URI from ecr
-    :param allowlist_for_daily_scans: list of allowlisted vulnerabilities for the image
+    :param image_sha: str, unique identifier of the image
+    :param upload_data: str, data that can be uploaded to the s3 object
+    :param s3_filename: name of the file upload to on image-data-storage bucket
     """
-    image_sha = get_sha_of_an_image_from_ecr(
-        ecr_client_for_enhanced_scanning_repo, ecr_enhanced_repo_uri
+    upload_data = json.dumps(
+        upload_data,
+        indent=4,
+        cls=EnhancedJSONEncoder,
     )
+    s3_filepath = f"{image_sha}/{s3_filename}"
     s3_resource = boto3.resource("s3")
     sts_client = boto3.client("sts")
     account_id = sts_client.get_caller_identity().get("Account")
-    s3object = s3_resource.Object(
-        f"image-data-storage-{account_id}", image_sha + "/ecr_allowlist.json"
+    s3object = s3_resource.Object(f"image-data-storage-{account_id}", s3_filepath)
+    s3object.put(Body=(bytes(upload_data.encode("UTF-8"))))
+    LOGGER.info(f"{s3_filename} uploaded to image-data-storage s3 bucket at {s3_filepath}")
+
+
+def upload_allowlist_and_core_packages_to_s3(
```

Comment:
This function has the potential to be renamed and can be designed in a slightly different way.

You can pass in a list of dictionaries, each of which has 2 keys `s3_location_within_image_data_storage_bucket` and `content`. Then iterate through the list to upload the elements one by one.

You can fetch the `allowlist_information` and `core_packages` information from outside of the method and pass in the list itself.